### PR TITLE
feat: remove wagmi/provider

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

... and use the escape-hatched viem instead: https://viem.sh/ https://github.com/wagmi-dev/viem

Basically gives us the same small set of core functionalities we need, with actual functions and without importing react-specific magic.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Smallish but not none, ENS resolution could be borked

## Testing

- ENS validation on sends is still working
- ENS name is still shown on send confirm step
- ENS name is still displayed on the user menu
- CI is happy

### Engineering

- Diff is red

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4304** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4304" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4304?utm_source=stack-comment).

File: src/lib/viem-client.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { mainnet } from '@wagmi/chains'
```
Comment (gomesalexandre) [inline src/lib/viem-client.ts:1]: Yeah, this is supposed to remove wagmi, but tests are borked if we import `viewm/chains` - which just proxies `@wagmi/chains` as a transient dep

File: src/components/Modals/Receive/ReceiveInfo.tsx [hunk 2]
```diff
@@ -85,21 +77,15 @@ export const ReceiveInfo = ({ asset, accountId }: ReceivePropsType) => {
...
   useEffect(() => {
-    if (isEnsNameLoaded && ensName) {
-      setEnsReceiveAddress(ensName)
-    }
-  }, [ensName, isEnsNameLoaded])
+    if (asset.chainId !== KnownChainIds.EthereumMainnet || !receiveAddress) return
+    ;(async () => {
+      const ensName = await viemClient.getEnsName({ address: receiveAddress as Address })
+      setEnsName(ensName)
+    })()
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/Modals/Receive/ReceiveInfo.tsx]: a41b9fafdf9675c96c2929e499e91f36bba0ec6a

**Review by 0xApotheosis — APPROVED** [review]
Tested ENS and Unstoppable domain resolution, LGTM!

**Pull request merged**